test/core/integration/concrete/lockup/transfer-from/transferFrom.t.sol

```diff
@@ -1,15 +1,14 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity >=0.8.22 <0.9.0;
 
+import { IERC4906 } from "@openzeppelin/contracts/interfaces/IERC4906.sol";
+import { IERC721 } from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+
 import { Errors } from "src/core/libraries/Errors.sol";
-import { Integration_Test } from "./../../../Integration.t.sol";
-import { Lockup_Integration_Shared_Test } from "./../../../shared/lockup/Lockup.t.sol";
 
-abstract contract TransferFrom_Integration_Concrete_Test is Integration_Test, Lockup_Integration_Shared_Test {
-    function setUp() public virtual override(Integration_Test, Lockup_Integration_Shared_Test) {
```

Comment:
it is not reverting because the first logic executed is `_update` function (which check for transferability)

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/f96237308f6d25ef69bd4c153e5651794651011f/contracts/token/ERC721/ERC721.sol#L143-L146

IMO it is irelevant who the caller is if the first check is reverting correctly, does it make sense?

```mermaid
sequenceDiagram
    participant Caller
    participant BurnFunction

    Caller ->> BurnFunction: call burn function
    BurnFunction ->> BurnFunction: 1. Check transferability
    BurnFunction ->> Caller: 2. Check caller
    BurnFunction ->> BurnFunction: 3. Burn
```

Reply:
This is an example of black box vs white box testing. You could argue that because you know the internal logic of `transferFrom`. On the other hand, without knowing the internal logic of `transferFrom` but just the natspecs of it, how would you have tested it otherwise?

IMO in that case, the first intuitive check would be to see who the caller is. And therefore it would become relevant who is the caller even before the first check. I would recommend to use black box testing approach where we test based on how the function should behave without getting so much into the internal logic. Having said that, the caller should be recipient for the whole test suite.

Reply:
since we are talking about only 1 test, i will prank the caller directly in the test function
and we can furthur discuss about black box vs white box testing later so that we can come up with a architecture

Reply:
btw, contradictory comment: https://github.com/sablier-labs/v2-core/pull/1058#discussion_r1819812368

Reply:
How is it contradictory?